lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -58,13 +94,63 @@ FModuleOp LowerLayersPass::buildNewModule(OpBuilder &builder, Location location,
   return newModule;
 }
 
-/// Process a module to remove any layer blocks it has.
-void LowerLayersPass::runOnModule(FModuleOp moduleOp) {
+void LowerLayersPass::removeLayersFromValue(Value value) {
+  auto type = dyn_cast<RefType>(value.getType());
+  if (!type || !type.getLayer())
+    return;
+  value.setType(type.removeLayer());
```

Comment:
Nice use of `setType`. This is the most efficient way to do this (as opposed to rewriting the type) as long it's used carefully.